src/session-pool.ts

```diff
@@ -1030,10 +1088,16 @@ export class SessionPool extends EventEmitter implements SessionPoolInterface {
    * @param {object} options The transaction options.
    */
   _prepareTransaction(session: Session): void {
+    const span = getActiveOrNoopSpan();
+    span.addEvent('creating transaction for session');
```

Comment:
this actually does not create a transaction. Its just creates a transaction object. transaction.id will always be null here. 

Remove both the events here

Reply:
In database class we have `transaction.begin` calls, there we should add these events. 

We can simply add these in transaction.begin() method.